### PR TITLE
Add the KFP v2 example and update the tutorial

### LINKED ISSUE #1586: Add a kubeflow example using KFP v2
Describe your problem
---------------------

As more users migrate the KFP v1 to KFP v2, the current kubeflow integration should support both KFP v1 and KFP v2. 

Additional context
------------------

A question from a kubeflow user.

> Have you had success using it in KFP v2? If so, did you use [kfp-kubernetes](https://kfp-kubernetes.readthedocs.io/en/kfp-kubernetes-1.0.0/#persistentvolumeclaim-dynamically-create-pvc-mount-then-delete) for volume mounts?

### PR DESCRIPTION
**Pull request by dashanji**
What do these changes do?
-------------------------

* Add the KFP v2 benchmark test.
* Update the tutorial.

Related issue number
--------------------

Fixes #1586

### CODE REVIEW

File: docs/tutorials/kubernetes/efficient-data-sharing-in-kubeflow-with-vineyard-csi-driver.rst [hunk 1]
```diff
@@ -277,6 +400,32 @@ As a result, we can find that with vineyard, the argo workflow
...
+The execution time of the pipeline shown in the kubeflow dashboard is as follows:
+
++------------+------------------+---------------+
+| data scale | without vineyard | with vineyard |
++============+==================+===============+
+| 8500 Mi    | 252s             | 228s          |
++------------+------------------+---------------+
+| 12000 Mi   | 299s             | 265s          |
++------------+------------------+---------------+
+| 15000 Mi   | 354s             | 316s          |
```
Comment (sighingnow) [inline docs/tutorials/kubernetes/efficient-data-sharing-in-kubeflow-with-vineyard-csi-driver.rst:420]: Why kfp v2 ran much slower than kfp v1?

Comment (dashanji) [reply docs/tutorials/kubernetes/efficient-data-sharing-in-kubeflow-with-vineyard-csi-driver.rst:420]: The kfp v2 will set the `image_pull_policy` to `Always``by default, and the kfp v1 is `IfNotPresent`. Pulling docker images will slow down the process.

Comment (dashanji) [reply docs/tutorials/kubernetes/efficient-data-sharing-in-kubeflow-with-vineyard-csi-driver.rst]: Corrected here: Different with the KFP V1, the KFP V2 will compile the pipeline to the IR YAML,
and the IR YAML will be submitted to the kubeflow server. Before each component of the
pipeline is executed, the kubeflow server will create a preprocess container to parse
the IR YAML and generate the relevant kubernetes resources. That means the kubeflow server
will create more pods than the previous argo workflow, thereby slowing down the execution 
speed of the entire pipeline.

**Comment by dashanji** [comment]
Hi @sighingnow, I tested the kfp v1 and kfp v2 again and reduced the affect of file cache, pulling image and scheduling etc. I think the current result is more convincing. Could you please take another review? Thanks.

**Review by sighingnow — APPROVED** [review]

**Comment by sighingnow** [comment]
> I think the current result is more convincing. Could you please take another review? Thanks.

Merged.

**Pull request merged**